addon/index.ts

```diff
@@ -156,28 +154,26 @@ function applyOptions(
   options: TaskGroupOptions | TaskOptions,
   task: TaskGroupProperty | TaskProperty
 ): (TaskGroupProperty | TaskProperty) & Decorator {
-  return Object.entries(options).reduce(
-    (
-      taskProperty,
-      [key, value]: [
-        keyof typeof options,
-        ObjectValues<Required<typeof options>>
-      ]
-    ) => {
-      assert(
-        `ember-concurrency-decorators: Option '${key}' is not a valid function`,
-        typeof taskProperty[key] === 'function'
-      );
-      if (value === true) {
-        return (taskProperty[key] as () => typeof taskProperty)();
-      }
-      return (taskProperty[key] as (o: typeof value) => typeof taskProperty)(
-        value
-      );
-    },
-    task
-    // The CP decorator gets executed in `createDecorator`
-  ) as typeof task & Decorator;
+  const keys = Object.keys(options) as Array<keyof typeof options>;
+
+  for (const key of keys) {
+    const value = options[key];
+
+    assert(
+      `ember-concurrency-decorators: Option '${key}' is not a valid function`,
+      typeof task[key] === 'function'
+    );
+
+    // maxConcurrency requires a numeric argument
+    if (value === true && key !== 'maxConcurrency') {
```

Comment:
curious about the special case here for `maxConcurrency`. Is it just because it's numeric or because it requires an argument? If so, there are other options that require an argument: `on`, `cancelOn`, `observes` too.

Reply:
I can remove the special casing since I resorted to using explicit type overrides for `task[key]`. Without those overrides, the first case errors in TS because not all methods that we might be calling allow no arguments (specifically `maxConcurrency`).

Reply:
Sorry for being overly vague there. Ultimately, I think we shouldn't _need_ to cast `task[key]` like we were doing, but I preserved it because I ran out of time.

This:

```js
    if (value === true) {
      (task[key] as () => typeof task)();
    }

    (task[key] as (o: typeof value) => typeof task)(value);
```

_Should be_:

```js
    if (value === true) {
      task[key]();
    }

    task[key](value);
```

Reply:
This _might_ be fixed by bringing back the `OptionValues` though, tesing now.

Reply:
Nice, yeah that makes sense!